projects/sunbird-video-player/src/lib/components/video-player/video-player.component.ts

```diff
@@ -72,22 +77,61 @@ export class VideoPlayerComponent implements AfterViewInit {
   play() {
     this.player.play();
     this.showPauseButton = true;
+    this.showPlayButton = false;
+    this.showBackwardButton = true;
+    this.showForwardButton = true;
+    this.viewerService.raiseHeartBeatEvent('PLAY');
   }
 
   pause() {
     this.player.pause();
+    this.showPauseButton = false;
     this.showPlayButton = true;
+    this.showBackwardButton = false;
+    this.showForwardButton = false;
+    this.viewerService.raiseHeartBeatEvent('PAUSE');
+  }
+
+  backward(time) {
+    this.player.currentTime(this.player.currentTime() - time);
+    this.viewerService.raiseHeartBeatEvent('BACKWARD');
+  }
+
+  forward(time) {
+    this.player.currentTime(this.player.currentTime() + time);
+    this.viewerService.raiseHeartBeatEvent('FORWARD');
   }
 
   handleVideoControls({type}) {
-    // showBackwardButton = false;
-    // showForwardButton = false;
-    // showPlayButton = true;
-    // showPauseButton = false;
     if(type === "playing") {
       this.showPlayButton = false;
       this.showPauseButton = true;
-      console.log(this.player.currentTime(), this.player.duration())
+      this.showBackwardButton = true;
+      this.showForwardButton = true;
+    }
+    if (type === 'ended') {
+      this.viewerService.visitedlength = this.player.currentTime();
```

Comment:
This is not correct visited length can be more than the current time since the user may click forward and backward